mbs_results/calculate_predicted_unit_value.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+
+import pandas as pd
+
+df = pd.read_csv(Path("tests") / "data" / "winsorisation" / "predicted_unit_value.csv")
+
+
+def calculate_predicted_unit_value(
+    df: pd.DataFrame,
+    period: str,
+    strata: str,
+    aux: str,
+    sampled: str,
+    a_weight: str,
+    target_variable: str,
+    nw_ag_flag: str,
+) -> pd.DataFrame:
+    """
+    Calculate link between target_variable and predictive_variable by strata,
+    a match_col must be supplied which indicates if target_variable
+    and predictive_variable can be linked.
+
+    Parameters
+    ----------
+    df : pd.Dataframe
+        Original dataframe.
+    period : str
+        Column name containing time period.
+    strata : str
+        Column name containing strata information (sic).
+    aux : str
+        Column name containing auxiliary variable (x).
+    sampled : str
+        Column name indicating whether it was sampled or not -boolean.
+    a_weight : str
+        Column name containing the design weight.
+    target_variable : str
+        Column name of the predicted target variable.
+    nw_ag_flag: str
+        column name indicating whether it can't be winsorised-
+        boolean (1 means it can't be winsorised, 0 means it can).
+    Returns
+    -------
+    df : pd.DataFrame
+        A pandas DataFrame with a new column containing the predicted unit value.
+    """
+
+    df = df.loc[(df["sampled"] == 1) & (df["nw_ag_flag"] == 0)]
+    df = df.reset_index(drop=True)
```

Comment:
Just curious, why do you need to reset and drop the index? I'm just thinking that other functions have set the index to be `period` and `date` columns and resetting might cause issues in the future. This might just be something to make a note of and see if it does lead to issues later :)